### PR TITLE
Update SDK Coders to return the Empty List from getCoderArguments

### PR DESCRIPTION
**Pull request by tgroh**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`.
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---
Brings existing Coders in line with the documentation introduced in #3032

### CODE REVIEW

**Comment by tgroh** [comment]
R: @kennknowles

File: sdks/java/core/src/main/java/org/apache/beam/sdk/util/WindowedValue.java [hunk 1]
```diff
@@ -685,9 +685,17 @@ public void registerByteSizeObserver(WindowedValue<T> value,
...
+     *
+     * @return a singleton list containing the {@code valueCoder} of this
+     *         {@link FullWindowedValueCoder}.
+     */
     @Override
     public List<? extends Coder<?>> getCoderArguments() {
-      return null;
+      // The value type is the only generic type parameter exposed by this coder. The component
+      // coders include the window coder as well
+      return Collections.singletonList(valueCoder);
```
Comment (kennknowles) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/util/WindowedValue.java]: Nice, and a bugfix.

**Comment by lukecwik** [comment]
Cherry pick?

**Comment by kennknowles** [comment]
Spec on the release branch does not allow nulls. So 2.0.0 is shipping with a known bug, but it is not "recall" severity. We can incorporate this bugfix into 2.1.0.

**Pull request closed without merging**